src/app/forms/input/input.html

```diff
@@ -0,0 +1,356 @@
+<!--
+  ~ Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+  ~ This software is released under MIT license.
+  ~ The full license information can be found in LICENSE in the root directory of this project.
+  -->
+
+
+<h1>Layout - {{layout}} - {{grid ? 'grid' : 'no grid'}}</h1>
+
+<form class="clr-form clr-form-{{layout}}">
+    <div class="clr-form-control" [ngClass]="{row: grid}">
+        <label for="{{layout}}-basic" class="clr-control-label" [ngClass]="{'col-xs-12': grid, 'col-lg-2': grid && layout !== 'vertical'}">
+            Basic input
+        </label>
+        <div class="clr-control-container" [ngClass]="{'col-xs-12': grid, 'col-lg-10': grid && layout !== 'vertical'}">
+            <div class="clr-input-wrapper">
+                <input type="text" id="{{layout}}-basic" placeholder="Enter value here" class="clr-input">
+                <clr-icon class="clr-validate-icon" shape="exclamation-circle"></clr-icon>
+            </div>
+            <span class="clr-subtext">Helper Text</span>
+        </div>
+    </div>
+
+    <div class="clr-form-control clr-form-control-disabled" [ngClass]="{row: grid}">
+        <label for="{{layout}}-disabled" class="clr-control-label" [ngClass]="{'col-xs-12': grid, 'col-lg-2': grid && layout !== 'vertical'}">
+            Basic input, disabled
+        </label>
+        <div class="clr-control-container" [ngClass]="{'col-xs-12': grid, 'col-lg-10': grid && layout !== 'vertical'}">
+            <div class="clr-input-wrapper">
+                <input type="text" id="{{layout}}-disabled" placeholder="Enter value here" class="clr-input" value="test value" disabled>
+                <clr-icon class="clr-validate-icon" shape="exclamation-circle"></clr-icon>
+            </div>
+            <span class="clr-subtext">Helper Text</span>
+        </div>
+    </div>
+
+    <div class="clr-form-control" [ngClass]="{row: grid}">
+        <label for="{{layout}}-basic-error" class="clr-control-label" [ngClass]="{'col-xs-12': grid, 'col-lg-2': grid && layout !== 'vertical'}">
+            Basic input, error</label>
+        <div class="clr-control-container clr-error" [ngClass]="{'col-xs-12': grid, 'col-lg-10': grid && layout !== 'vertical'}">
+            <div class="clr-input-wrapper">
+                <input type="text" id="{{layout}}-basic-error" placeholder="Enter value here" class="clr-input">
+                <clr-icon class="clr-validate-icon" shape="exclamation-circle"></clr-icon>
+            </div>
+            <span class="clr-subtext">Helper Text</span>
+        </div>
+    </div>
+
+    <div class="clr-form-control" [ngClass]="{row: grid}">
+        <label class="clr-control-label" [ngClass]="{'col-xs-12': grid, 'col-lg-2': grid && layout !== 'vertical'}">
+            Basic checkbox</label>
+        <div class="clr-control-container" [ngClass]="{'col-xs-12': grid, 'col-lg-10': grid && layout !== 'vertical'}">
+            <div class="clr-checkbox-wrapper">
```

Comment:
Are we breaking the checkboxes too? The current checkboxes have this markup:

```
<div class="checkbox">
      <input type="checkbox" id="checkrads_1">
      <label for="checkrads_1">Checkbox 1</label>
</div>
```

The new ones though have `.clr-checkbox-wrapper`, wrapping multiple checkboxes and each individual checkbox not having a wrapper around it.

Is the reason for this change simplicity of the markup?

Reply:
Are the new forms compatible with the old style of the markup? @youdz's implementation of the Angular component still has a wrapper around each individual checkbox: https://github.com/vmware/clarity/blob/master/src/clr-angular/forms/checkbox/checkbox-container.spec.ts#L17

Reply:
I've looking into this a little bit, but need to do more prototyping to verify how they work together.